fog/view/server/src/db_fetcher.rs

```diff
@@ -503,95 +468,146 @@ mod tests {
         assert_eq!(fetched_records.len(), blocks_and_records.len());
 
         for (i, fetched_record) in fetched_records.iter().enumerate() {
-            assert_eq!(fetched_record.ingest_invocation_id, invoc_id1);
-            assert_eq!(fetched_record.block_index, i as u64 + 10);
+            assert_eq!(fetched_record.ingress_key, ingress_key);
+            assert_eq!(fetched_record.block_index, i as u64 + 20);
             assert_eq!(blocks_and_records[i].1, fetched_record.records);
         }
 
         assert!(db_fetcher.get_pending_fetched_records().is_empty()); // The previous call should have drained this
 
+        sleep(Duration::from_millis(100));
+
+        let ingress_keys = db_fetcher.get_highest_processed_block_context();
+        assert_eq!(
+            ingress_keys,
+            vec![IngressPublicKeyRecord {
+                key: ingress_key.clone(),
+                status: IngressPublicKeyStatus {
+                    start_block: 10,
+                    pubkey_expiry: 0,
+                    retired: false,
+                    lost: false,
+                },
+                last_scanned_block: Some(29),
+            }]
+        );
+
         // Add more blocks but this time leave a hole between the previous blocks and
         // the new ones. They should not get picked up untill a missed blocks
         // range is reported.
-        let mut blocks_and_records = Vec::new();
-        for i in 30..40 {
+        let mut blocks_and_records_40_50 = Vec::new();
+        for i in 40..50 {
             let (block, records) = fog_test_infra::db_tests::random_block(&mut rng, i, 5); // 5 outputs per block
 
             db.add_block_data(&invoc_id1, &block, 0, &records).unwrap();
-            blocks_and_records.push((block, records));
+            blocks_and_records_40_50.push((block, records));
         }
 
         sleep(Duration::from_secs(1)); // Supposedly enough time for at least some blocks to get picked up.
 
         assert!(db_fetcher.get_pending_fetched_records().is_empty());
 
-        // Report a missing block range.
-        db.report_missed_block_range(&BlockRange::new(20, 30))
-            .unwrap();
+        sleep(Duration::from_millis(100));
+
+        let ingress_keys = db_fetcher.get_highest_processed_block_context();
+        assert_eq!(
+            ingress_keys,
+            vec![IngressPublicKeyRecord {
+                key: ingress_key.clone(),
+                status: IngressPublicKeyStatus {
+                    start_block: 10,
+                    pubkey_expiry: 0,
+                    retired: false,
+                    lost: false,
+                },
+                last_scanned_block: Some(49),
```

Comment:
This was initially confusing to me, but the 49 value comes from MAX(block_index) on the processed blocks db table, which makes sense since we did add blocks [40-50) above. I would document this for future us.

Reply:
done